tee-worker/litentry/pallets/identity-management/src/tests.rs

```diff
@@ -422,16 +422,53 @@ fn get_id_graph_works() {
 		let id_graph = IMT::get_id_graph(&who);
 		assert_eq!(id_graph.len(), 22);
 
-		// despite the first and second items from the graph have the same `link_block` number the prime identity goes first
-		assert_eq!(id_graph.get(0).unwrap().1.link_block, id_graph.get(1).unwrap().1.link_block);
 		assert_eq!(id_graph.get(0).unwrap().0, who);
-		assert_eq!(id_graph.get(1).unwrap().0, alice_twitter_identity(1));
 
 		// index 21 has the newest identity
 		assert_eq!(id_graph.get(21).unwrap().0, alice_twitter_identity(21));
 	});
 }
 
+#[test]
+fn get_id_graph_identities_within_same_block() {
+	new_test_ext().execute_with(|| {
+		let who: Identity = ALICE.into();
+		System::set_block_number(1);
+
+		let mut identities = vec![
+			(alice_twitter_identity(1), vec![]),
+			(alice_substrate_identity(), vec![Web3Network::LitentryRococo]),
+			(alice_evm_identity(), vec![Web3Network::Ethereum]),
+			(bob_substrate_identity(), vec![Web3Network::Litentry]),
+		];
+
+		let mut rng = rand::thread_rng();
+		identities.shuffle(&mut rng);
```

Comment:
Why such shuffle is needed ? Maybe you could declare identities in desired order ?
I think the outcome of shuffle can be the same order as before.

Reply:
The intention was to show that any order is possible. And initially I've implemented it with two different orders for the identities, and then I realised that I could refactor it in a more dynamic way. Do you think it's unnecessary to have such approach?

Reply:
I think the fix order is enough and solid to verify whether you get the expected order later.

Reply:
If all random orders are equal just pick one, if not then write as many test cases as comes to your mind.
Having static inputs for tests results in deterministic outputs.